src/activity/activity.module.ts

```diff
@@ -240,6 +241,19 @@ export class ActivityModuleImpl implements ActivityModule {
       return ctx;
     } catch (error) {
       this.events.emit("errorInitializingActivity", activity, error);
+      if (options?.signalOrTimeout instanceof AbortSignal && options.signalOrTimeout.aborted) {
+        const error =
+          options.signalOrTimeout.reason.name === "TimeoutError"
+            ? new GolemTimeoutError(
+                "Initializing of the exe-unit has been aborted due to a timeout",
+                options.signalOrTimeout.reason,
+              )
+            : new GolemAbortError(
+                "Initializing of the exe-unit has been aborted by a signal",
+                options.signalOrTimeout.reason,
+              );
+        throw error;
+      }
```

Comment:
Why is this here? `ctx.before` already throws `GolemAbortError`

Reply:
not all parts of `ctx.before` throw an `GolemAbortError` in this case. Some throw `GolemWorkError` with interrupted script execution in activity. It is true that in the `previous` field there will be AbortError from Activity, but I wanted to avoid inconsistency.

Reply:
ok, but in that case if someone passes a timeout (number) then this check won't run, it should be consistent. Have you considered checking the `ctx.abortSignal` directly?

Reply:
Yes, you are right. I moved the handling of this to `ctx.before` in the work-context and refactored it a bit. See now.